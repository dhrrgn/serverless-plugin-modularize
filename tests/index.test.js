import Modularize from '../src/index';
import fs from 'fs';

const createServerless = ({ custom } = {}) => ({
  service: {
    custom: custom || {
      foo: 'bar',
    },
  },
  cli: {
    consoleLog: jest.fn(),
  },
  utils: {
    readFileSync: fs.readFileSync 
  }
});

const createModularize = (sls) => {
  const plugin = new Modularize(sls);
  plugin.mergeModules();
  return plugin;
};

describe('Modularize', () => {
  it('ignored when missing globs', () => {
    const serverless = createServerless();
    createModularize(serverless);

    expect(serverless).toMatchSnapshot();
  });

  it('merges external files', () => {
    const serverless = createServerless({
      custom: {
        modularize: {
          glob: '**/example.json',
        },
      },
    });
    createModularize(serverless);
    expect(serverless).toMatchSnapshot();
  });
});
