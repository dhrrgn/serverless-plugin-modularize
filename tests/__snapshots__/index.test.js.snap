// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modularize ignored when missing globs 1`] = `
Object {
  "cli": Object {
    "consoleLog": [MockFunction],
  },
  "service": Object {
    "custom": Object {
      "foo": "bar",
    },
    "functions": Object {},
    "plugins": Array [],
    "provider": Object {},
    "resources": Object {},
  },
  "utils": Object {
    "readFileSync": [Function],
  },
}
`;

exports[`Modularize merges external files 1`] = `
Object {
  "cli": Object {
    "consoleLog": [MockFunction],
  },
  "service": Object {
    "custom": Object {
      "modularize": Object {
        "glob": "**/example.json",
      },
    },
    "functions": Object {},
    "plugins": Array [],
    "provider": Object {
      "environment": Object {
        "FOO": "BAR",
      },
    },
    "resources": Object {},
  },
  "utils": Object {
    "readFileSync": [Function],
  },
}
`;
